### PR TITLE
feat: remove azure account extension depends

### PR DESCRIPTION
**Pull request by wenytang-ms**
image
Animation

### CODE REVIEW

File: src/utils/utils.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+export namespace Utils {
```
Comment (formulahendry) [inline src/utils/utils.ts:4]: Is it a general utils or only for Azure?
If it is a general utils, then merge with `generalUtils.ts`.
If it is only for Azure, better rename it.

File: src/azure/azureLogin/azureAuth.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    }
+
+    export function getScopes(tenantId: string | null, options: GetAuthSessionOptions): string[] {
+        const defaultScopes = options.scopes || [AzureAuth.getDefaultScope(AzureAuth.getConfiguredAzureEnv().resourceManagerEndpointUrl)];
+        const tenantScopes = tenantId ? [`VSCODE_TENANT:${tenantId}`] : [];
+        const clientIdScopes = options.applicationClientId ? [`VSCODE_CLIENT_ID:${options.applicationClientId}`] : [];
+        return [...defaultScopes, ...tenantScopes, ...clientIdScopes];
+    }
+
+    type AuthProviderId = "microsoft" | "microsoft-sovereign-cloud";
```
Comment (formulahendry) [inline src/azure/azureLogin/azureAuth.ts:141]: what is microsoft-sovereign-cloud?

Comment (wenytang-ms) [reply src/azure/azureLogin/azureAuth.ts:141]: azure subscriptions for special customers, such as azure china, azure government and so on.
https://blogs.microsoft.com/blog/2023/12/14/microsoft-cloud-for-sovereignty-now-generally-available-opening-new-pathways-for-government-innovation/

File: src/utils/utils.ts (hunk 1 above)
Comment (wenytang-ms) [reply src/utils/utils.ts]: merged into generalUtils.ts

File: resources/azureAccount.svg
Comment (formulahendry) [inline resources/azureAccount.svg:1]: Is this file used?

File: src/azure/azureLogin/azureAuth.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        if (!AzureAuth.isReady(sessionProvider)) {
+            return { succeeded: false, error: "Not signed in." };
+        }
+        return { succeeded: true, result: sessionProvider };
+    }
+
+    async function waitForSignIn(sessionProvider: AzureSessionProvider): Promise<void> {
+        const options: vscode.ProgressOptions = {
+            location: vscode.ProgressLocation.Notification,
+            title: "Waiting for sign-in",
```
Comment (formulahendry) [inline src/azure/azureLogin/azureAuth.ts:78]: put into uiStrings.ts

File: src/azure/azureLogin/azureAuth.ts [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        return endpointUrl.endsWith("/") ? `${endpointUrl}.default` : `${endpointUrl}/.default`;
+    }
+
+    export async function quickPickTenant(tenants: Tenant[]): Promise<Tenant | undefined> {
+        const items: (vscode.QuickPickItem & { tenant: Tenant })[] = tenants.map((t) => ({
+            label: `${t.name} (${t.id})`,
+            tenant: t,
+        }));
+        const result = await vscode.window.showQuickPick(items, {
+            placeHolder: "Select a tenant",
```
Comment (formulahendry) [inline src/azure/azureLogin/azureAuth.ts:40]: put into uiStrings.ts

File: src/azure/azureLogin/azureAuth.ts [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+    export function getEnvironment(): Environment {
+        return getConfiguredAzureEnv();
+    }
+
+    export function getCredential(sessionProvider: ReadyAzureSessionProvider): TokenCredential {
+        return {
+            getToken: async () => {
+                const session = await sessionProvider.getAuthSession();
+                if (GeneralUtils.failed(session)) {
+                    throw new Error(`No Microsoft authentication session found: ${session.error}`);
```
Comment (formulahendry) [inline src/azure/azureLogin/azureAuth.ts:20]: put into uiStrings.ts

File: src/azure/azureLogin/azureAuth.ts [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+            return Environment.ChinaCloud;
+        } else if (environmentSettingValue === "USGovernment") {
+            return Environment.USGovernment;
+        } else if (environmentSettingValue === "custom") {
+            const customCloud = authProviderConfig.get<EnvironmentParameters | undefined>("customEnvironment");
+            if (customCloud) {
+                return new Environment(customCloud);
+            }
+
+            throw new Error(
```
Comment (formulahendry) [inline src/azure/azureLogin/azureAuth.ts:114]: put into uiStrings.ts

File: src/azure/azureLogin/azureSessionProvider.ts [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+         * an error message is returned.
+         */
+        public async getAuthSession(options?: GetAuthSessionOptions): Promise<GeneralUtils.Errorable<AzureAuthenticationSession>> {
+            await this.initializePromise;
+            if (this.signInStatusValue !== SignInStatus.SignedIn) {
+                return { succeeded: false, error: `Not signed in (${this.signInStatusValue}).` };
+            }
+
+            if (this.tenants.length === 0) {
+                return { succeeded: false, error: "No tenants found." };
```
Comment (formulahendry) [inline src/azure/azureLogin/azureSessionProvider.ts:160]: put into uiStrings.ts

File: src/azure/azureLogin/azureSessionProvider.ts [hunk 8]
```diff
@@ -0,0 +1,272 @@
...
+            }
+
+            if (!this.selectedTenantValue) {
+                if (this.tenants.length > 1) {
+                    const selectedTenant = await AzureAuth.quickPickTenant(this.tenants);
+                    if (!selectedTenant) {
+                        return { succeeded: false, error: "No tenant selected." };
```
Comment (formulahendry) [inline src/azure/azureLogin/azureSessionProvider.ts:167]: put into uiStrings.ts

File: src/azure/azureLogin/azureSessionProvider.ts [hunk 9]
```diff
@@ -0,0 +1,272 @@
...
+                    // The 'silent' option is incompatible with most other options, so we completely replace the options object here.
+                    session = await authentication.getSession(AzureAuth.getConfiguredAuthProviderId(), scopes, { silent: true });
+                }
+
+                if (!session) {
+                    session = await authentication.getSession(AzureAuth.getConfiguredAuthProviderId(), scopes, options);
+                }
+
+                if (!session) {
+                    return { succeeded: false, error: "No Azure session found." };
```
Comment (formulahendry) [inline src/azure/azureLogin/azureSessionProvider.ts:260]: put into uiStrings.ts

File: src/azure/azureLogin/azureSessionProvider.ts [hunk 10]
```diff
@@ -0,0 +1,272 @@
...
+                }
+
+                return { succeeded: true, result: Object.assign(session, { tenantId }) };
+            } catch (e) {
+                return { succeeded: false, error: `Failed to retrieve Azure session: ${GeneralUtils.getErrorMessage(e)}` };
```
Comment (formulahendry) [inline src/azure/azureLogin/azureSessionProvider.ts:265]: put into uiStrings.ts

**Review by formulahendry — CHANGES_REQUESTED** [review]
Overall looks good. Just need to check if there are any more strings needed to put into `uiStrings.ts`

File: resources/azureAccount.svg
Comment (wenytang-ms) [reply resources/azureAccount.svg:1]: removed

**Comment by wenytang-ms** [comment]
> Overall looks good. Just need to check if there are any more strings needed to put into `uiStrings.ts`

@formulahendry The ui string all updated, please review again

File: src/test/unit/azure/azureLogin/azureAuth.test.ts [hunk 11]
```diff
@@ -0,0 +1,211 @@
...
+        const res = AzureAuth.getConfiguredAzureEnv();
+        assert.strictEqual(res.name, "AzureChinaCloud");
+    });
+    it('getConfiguredAzureEnv return USGovernment', async () => {
+        sandbox.stub(workspaceConfiguration, "get").returns('USGovernment');
+        sandbox.stub(vscode.workspace, "getConfiguration").returns(workspaceConfiguration);
+        const res = AzureAuth.getConfiguredAzureEnv();
+        assert.strictEqual(res.name, "AzureUSGovernment");
+    });
+    it('getConfiguredAzureEnv return USGovernment', async () => {
```
Comment (formulahendry) [inline src/test/unit/azure/azureLogin/azureAuth.test.ts:200]: Not 'getConfiguredAzureEnv return USGovernment'?

Comment (wenytang-ms) [reply src/test/unit/azure/azureLogin/azureAuth.test.ts]: updadted.

**Review by formulahendry — APPROVED** [review]

**Pull request merged**